src/main/java/io/cdap/plugin/http/source/batch/HttpBatchSource.java

```diff
@@ -62,10 +65,14 @@ public HttpBatchSource(HttpBatchSourceConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
-    config.validate(failureCollector); // validate when macros not yet substituted
-    config.validateSchema();
-
-    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    config.validate(failureCollector);
+    Schema schema = config.getSchema();
+    FormatContext context = new FormatContext(failureCollector, null);
+    if (schema == null && !config.containsMacro("schema")) {
+      HttpInputFormatProvider httpInputFormatProvider = new HttpInputFormatProvider(config);
+      schema = httpInputFormatProvider.getSchema(context);
```

Comment:
what is the use of format context here, can't we use failure collector

Reply:
Yes we can, i have removed `format context` , this was being used as this was taken from Abstract File Source.